app/src/main/java/ch/epfl/sdp/drone3d/service/impl/location/AndroidLocationService.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2021  Drone3D-Team
+ * The license can be found in LICENSE at root of the repository
+ */
+
+package ch.epfl.sdp.drone3d.service.impl.location
+
+import android.Manifest
+import android.content.Context
+import android.content.pm.PackageManager
+import android.location.Criteria
+import android.location.LocationListener
+import android.location.LocationManager
+import ch.epfl.sdp.drone3d.service.api.location.LocationService
+import ch.epfl.sdp.drone3d.service.module.LocationModule.LocationProvider
+import com.mapbox.mapboxsdk.geometry.LatLng
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Location service using android location functionalities
+ */
+@Singleton
+class AndroidLocationService @Inject constructor(
+    private val locationManager: LocationManager,
+    @LocationProvider val locationProvider: String?,
+    private val locationCriteria: Criteria,
+    @ApplicationContext private val context: Context
+) : LocationService {
+
+    private var listenerId = 1
+    private val listeners: MutableMap<Int, LocationListener> = HashMap()
+
+    private fun getMyLocationProvider(): String? {
+        return locationProvider ?: locationManager.getBestProvider(locationCriteria, true)
+    }
+
+    override fun isLocationEnabled(): Boolean {
+        val currentPermission =
+            context.checkSelfPermission(Manifest.permission.ACCESS_FINE_LOCATION)
+        return currentPermission == PackageManager.PERMISSION_GRANTED && getMyLocationProvider() != null
+    }
+
+    override fun getCurrentLocation(): LatLng? {
+        if (!isLocationEnabled()) {
+            return null
+        }
+        return try {
+            val provider = getMyLocationProvider() ?: return null
+            val loc =
+                locationManager.getLastKnownLocation(provider)
+                    ?: return null
+            LatLng(loc.latitude, loc.longitude)
+        } catch (ex: SecurityException) {
+            // We need to explicitly catch this exception that raises if permission is not granted
```

Comment:
If the exception is just rethrown, we should remove the try/catch block as it has no prupose.

Reply:
You're right, I could remove it. I will change it on another PR.